pkg/networkpath/traceroute/common/common.go

```diff
@@ -62,7 +62,28 @@ type (
 		InnerDstIP   net.IP
 		InnerSrcPort uint16
 		InnerDstPort uint16
-		InnerSeqNum  uint32
+		// InnerIdentifier will be populated with
+		// an additional identifcation field for matching
+		// received packets. For TCP packets, this is the
+		// sequence number. For UDP packets, this is the
+		// checksum, a uint16 cast to a uint32.
+		InnerIdentifier uint32
+	}
+
+	ICMPParser struct {
```

Comment:
Can this be a little cleaner if we had
```go
type ICMParser interface ...
```
and then two different implementing types for the tcp and udp parsers?

Reply:
I separated this out into a new package `icmp` and made an interface